prymer/primer3/primer3_input.py

```diff
@@ -96,7 +100,8 @@ class Primer3Input:
     target: Span
     task: Primer3TaskType
     params: Primer3Parameters
-    weights: Primer3Weights = Primer3Weights()
+    primer_weights: Optional[PrimerAndAmpliconWeights] = PrimerAndAmpliconWeights()
+    probe_weights: Optional[ProbeWeights] = None
```

Comment:
**note** Per discussion, #48 adds validation of the presence of required weights, based on the specified task.